business/checkers/authorization/no_host_checker.go

```diff
@@ -50,9 +49,6 @@ func (n NoHostChecker) validateHost(ruleIdx int, to []*api_security_v1beta.Rule_
 		return nil, true
 	}
 	namespace, clusterName := n.AuthorizationPolicy.Namespace, n.AuthorizationPolicy.ClusterName
```

Comment:
I've found a regression on this one.
Using this example 
https://kiali.io/docs/features/validations/#kia0104---this-host-has-no-matching-entry-in-the-service-registry
the host detection doesn't seem to work well.

Reply:
the behavior in this PR is the same as in master, can you please provide more details what is wrong here?

all the validations are shown: 
![Screenshot from 2022-01-26 17-33-40](https://user-images.githubusercontent.com/604313/151205415-22d3c0b3-72d8-4c2b-a4cd-f2563902d972.png)

Reply:
I don't see the validations in my example:
![image](https://user-images.githubusercontent.com/1662329/151343128-df151a84-bc49-4aa7-bded-a38564cefa71.png)

I'll validate if the master branch is working on my side, but my expectation would be to see the same screenshot you've uploaded.

Reply:
In the master branch I have the same screenshot:
![image](https://user-images.githubusercontent.com/1662329/151344283-2556a69d-fe0d-409f-8ee5-318d9f6b56da.png)

Reply:
I'm using the "default" namespace, this is the tested configuration:
```
kind: AuthorizationPolicy
apiVersion: security.istio.io/v1beta1
metadata:
  name: httpbin
  namespace: default
  uid: 58a98524-8e7c-4047-9a39-dba7daf0472b
  resourceVersion: '7153'
  generation: 2
  creationTimestamp: '2022-01-27T10:37:16Z'
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: >
      {"apiVersion":"security.istio.io/v1beta1","kind":"AuthorizationPolicy","metadata":{"annotations":{},"name":"httpbin","namespace":"default"},"spec":{"rules":[{"from":[{"source":{"principals":["cluster.local/ns/default/sa/sleep"]}},{"source":{"namespaces":["default"]}}],"to":[{"operation":{"hosts":["wrong","ratings","details.default","reviews.default.svc.cluster.local","productpage.outside","google.com","google.org"],"methods":["GET","/package.service/method"],"paths":["/info*"]}},{"operation":{"methods":["POST"],"paths":["/data"]}}],"when":[{"key":"request.auth.claims[iss]","values":["https://accounts.google.com"]}]}],"selector":{"matchLabels":{"app":"httpbin","version":"v1"}}}}
  managedFields:
    - manager: kubectl-client-side-apply
      operation: Update
      apiVersion: security.istio.io/v1beta1
      time: '2022-01-27T10:37:16Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:kubectl.kubernetes.io/last-applied-configuration': {}
        'f:spec': {}
    - manager: pilot-discovery
      operation: Update
      apiVersion: security.istio.io/v1beta1
      time: '2022-01-27T10:37:17Z'
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          .: {}
          'f:conditions': {}
          'f:observedGeneration': {}
spec:
  rules:
    - from:
        - source:
            principals:
              - cluster.local/ns/default/sa/sleep
        - source:
            namespaces:
              - default
      to:
        - operation:
            hosts:
              - wrong
              - ratings
              - details.default
              - reviews.default.svc.cluster.local
              - productpage.outside
              - google.com
              - google.org
            methods:
              - GET
              - /package.service/method
            paths:
              - /info*
        - operation:
            methods:
              - POST
            paths:
              - /data
      when:
        - key: 'request.auth.claims[iss]'
          values:
            - 'https://accounts.google.com'
status:
  conditions:
    - type: Reconciled
      status: 'True'
      lastProbeTime: '2022-01-27T10:38:58.783262913Z'
      lastTransitionTime: '2022-01-27T10:38:58.783263133Z'
      message: 112/112 proxies up to date.
  observedGeneration: '2'
```

Reply:
@lucasponce Is the cache enabled on your local env? if it is disabled, registry does not provide any services or configs, so validations will not happen.

Reply:
No, cached is enabled as usual.

Reply:
tried on new environment, still cannot see the problem

Reply:
Ok, I guess I've found that in my scenario I kept having a VS with a wildcard:
![image](https://user-images.githubusercontent.com/1662329/151369328-a3c6bbf4-2de0-4873-9076-045ba2369790.png)

And that made that is recognized in the logic:
![image](https://user-images.githubusercontent.com/1662329/151369487-ebaf9d70-174f-46f0-81c9-c4d0200bc766.png)

So, I think that's valid. I'll update my env to continue with the review.

Reply:
Ok, that was the case:
![image](https://user-images.githubusercontent.com/1662329/151372895-d046d77c-4701-4d16-ae15-923154957ce7.png)